Validate SSL in panos_import

kbreit opened the pull request:
##### SUMMARY
Bug #36936 states `panos_import` should validate SSL connections to avoid man in the middle attacks.

Fixes #36936 
##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
panos_import

##### ANSIBLE VERSION
```
ansible 2.6.0 (panos/bug-36936 67c27ef877) last updated 2018/03/03 19:53:31 (GMT -500)
  config file = None
  configured module search path = ['/Users/kbreit/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/kbreit/Documents/Programming/ansible/lib/ansible
  executable location = /Users/kbreit/Documents/Programming/ansible/bin/ansible
  python version = 3.5.4 (default, Feb 25 2018, 14:56:02) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```


##### ADDITIONAL INFORMATION
This should likely be back ported to previous versions since the code seems to be the same.

No difference in output

kbreit committed:
Fix bug 36936

kbreit commented:
Note to reviewers. This changes default behavior from false to true. It is a much better default but may break existing plays.

kbreit committed:
Added version_added to argument and fixed whitespace

gundalow commented on lib/ansible/modules/network/panos/panos_import.py:
```diff
@@ -60,6 +63,12 @@
         default: None
+    validate_ssl:
+        description:
+            - Whether or not certificates should be validated
```
Please change to
If C(no), SSL certificates will not be validated. This should only set to no used on personally controlled sites using self-signed certificates.

gundalow commented on lib/ansible/modules/network/panos/panos_import.py:
```diff
@@ -60,6 +63,12 @@
+        description:
+            - Whether or not certificates should be validated
+        type: bool
+        default: True
```
Yes

kbreit committed:
Update panos_import documentation

Update parameter documentation and add note.

kbreit committed:
Add type documentation

kbreit reviewed (commented):
Changes applied

kbreit committed:
added version number for documentation

For real

kbreit commented:
All requested changes should be integrated. It's now just a question of whether we should change the default to validate SSL or if staying with 'false' as default makes sense.

kbreit commented:
I am going to change to have the default to false to not break backwards compatibility.

If there is still a merge conflict, I’ll need someone to help fix that.

kbreit committed:
Integrated recommended changes

- Added recommended changes from PR

kbreit committed:
Changed validate_ssl default back to True considering there is a
note at the top of documentation explaining change

gundalow commented on lib/ansible/modules/network/panos/panos_import.py:
```diff
@@ -47,13 +50,24 @@
     category:
         description:
             - Category of file uploaded. The default is software.
+
```
Please remove blank line

gundalow commented on lib/ansible/modules/network/panos/panos_import.py:
```diff
@@ -47,13 +50,24 @@
             - Category of file uploaded. The default is software.
+
+            - See API > Import section of the API reference for category options.
+        required: false
```
Please remove `required: false`

gundalow commented on lib/ansible/modules/network/panos/panos_import.py:
```diff
@@ -47,13 +50,24 @@
     url:
         description:
             - URL of the file that will be imported to device.
+        required: false
```
Please remove `required: false`

kbreit commented:
I've changed my mind and am defaulting this to `on`. Please review. If approved, I'm going to propagate this change to the other PAN modules.

kbreit committed:
Format changes based on recommendations from gundalow

kbreit reviewed (commented):
All changes integrated in latest push.

gundalow commented:
@ivanbojer @jtschichold You happy with this change?

ericwb commented on lib/ansible/modules/network/panos/panos_import.py:
```diff
@@ -47,13 +50,21 @@
         description:
             - URL of the file that will be imported to device.
+        default: None
+    validate_ssl:
```
"ssl" is a legacy, deprecated term for TLS. Other modules use "validate_certs" which might be better here.

kbreit committed:
Rename validate_ssl to validate_cert

kbreit committed:
Change description to remove SSL reference

kbreit reviewed (commented):
Makes sense. Changes made.

kbreit committed:
Change url default ih documentation

kbreit commented:
Can this be merged?

ericwb commented on lib/ansible/modules/network/panos/panos_import.py:
```diff
@@ -47,13 +50,20 @@
     url:
         description:
             - URL of the file that will be imported to device.
+    validate_cert:
```
To be consistent with other modules, this should be validate_certs, not validate_cert.

ericwb commented on lib/ansible/modules/network/panos/panos_import.py:
```diff
@@ -47,13 +50,20 @@
             - URL of the file that will be imported to device.
+    validate_cert:
+        description:
+            - If C(no), SSL certificates will not be validated. This should only set to no used on personally controlled sites using self-signed certificates.
```
"This should only set to no used on" - sounds awkward.  I also don't recommend reasons for when to turn this off.  I'd just warn them it's not recommend.   Even personally controlled sites using self-signed certs can fall victim to MITM attacks.

ericwb reviewed (changes requested):
Sorry, couple more changes please.  Thanks for your changes so far.

kbreit committed:
Integrated small changes from bug report
- Renamed validate_cert to validate_certs
- Changed documentation for disabling cert validation

kbreit reviewed (commented):
All changes applied and pushed.

ericwb reviewed (approved):
Looks good to me

gundalow commented:
Thanks, merged into `devel` will be releases in `2.6.0`
Not backporting as it introduces a new option.
